### DISCUSSION TITLE
Learn how to #BUIDL Web3 Games with Soroban smart contracts on Stellar #NFTs #Gaming

### DISCUSSION THREAD

**Discussion by FredericRezeau** [stellar/sorobounty-spectacular#5] [General]
Hi everyone,

Taking this opportunity to raise awareness on how the tech can serve use cases like web3 gaming, NFT and metaverse in general. Here are some educational resources showing developers how to get started with implementing quintessential web3 gaming use cases using Soroban smart contracts deployed on Stellar. Hope this can help bring more builders on the ecosystem.

* **soroban-snooker** A smart contract showcasing web3 gaming with Soroban on Stellar (game client code also available)
   [https://github.com/FredericRezeau/soroban-snooker](https://github.com/FredericRezeau/soroban-snooker)
* **Soroban Snooker Game** A live demo game client using the soroban-snooker contract
   [https://snooker.litemint.com](https://snooker.litemint.com)
* **Youtube video** Dev code walkthough
   [https://www.youtube.com/watch?v=hUvyr3XxHv0](https://www.youtube.com/watch?v=hUvyr3XxHv0)
   
   With these resources, developers can learn:
   
- [x] How to implement In-App Purchases to receive payments from players.
- [X] How to issue rewards to players such as achievement NFTs, as part of the gameplay experience.
- [X] How to implement decentralized gameplay validation, including pool physics and time-based checks.
- [X] How to utilize auth and set up admin functions for management of game assets and withdrawals.
- [X] How to employ different Soroban storage types to manage game data efficiently.
- [X] How to leverage custom types for client-contract communication.
- [X] How to integrate the Freighter Stellar wallet to the game client.
- [X] How to implement a local/guest mode for players without a wallet.

**Comment by briwylde08** [comment]
Amazing! Thank you so much, Fred! We’ll review your submission and be in contact with you within the next week. :tada:

**Comment by rice2000** [comment]
This is awesome.

**Reply by FredericRezeau** [reply]
Thank you! 🙏

**Comment by briwylde08** [comment]
Alright! We completed the README walkthrough and watched the video and everything looks awesome! [🎉](https://emojipedia.org/party-popper/)

Given our guidelines, your submission may qualify for an award of $1,250 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Reply by FredericRezeau** [reply]
Hi Bri,

This is awesome news!

Thank you so much for this and for the opportunity 🙏

All the best,
Fred

**Comment by briwylde08** [comment]
Hi Fred!

Thank you for providing the requested information! You’ve passed KYC and we’ve logged your award payment of $1,250 worth of XLM, which will be sent to the wallet you designated in the next week.

Thanks again and please let me know if you have any questions!

Best,
Bri

**Comment by FredericRezeau** [comment]
Thank you so much!
